### PR TITLE
chore: complete setting up auth middleware

### PR DESCRIPTION
**Pull request by tpluscode**

### CODE REVIEW

File: .local.env [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
 UI_BASE=https://app.cube-creator.lndo.site/
 
+# SPARQL endpoints
 STORE_QUERY_ENDPOINT=http://store:3030/cube-creator/query
 STORE_UPDATE_ENDPOINT=http://store:3030/cube-creator/update
 STORE_GRAPH_ENDPOINT=http://store:3030/cube-creator/data
 
-AUTH_AUDIENCE=https://datacube.zazukoians.org/
+# OIDC config
+AUTH_AUDIENCE=cube-creator
```
Comment (tpluscode) [inline .local.env:16]: Do we need a new client in Keycloak to have a proper audience here? @sandhose

File: apis/core/lib/auth.ts [hunk 2]
```diff
@@ -12,15 +14,15 @@ declare module 'express-serve-static-core' {
...
-const createJwtHandler = (credentialsRequired: boolean) => jwt({
+const createJwtHandler = (credentialsRequired: boolean, jwksUri: string) => jwt({
   // Dynamically provide a signing key
   // based on the kid in the header and
   // the signing keys provided by the JWKS endpoint.
   secret: jwksRsa.expressJwtSecret({
     cache: true,
     rateLimit: true,
     jwksRequestsPerMinute: 5,
-    jwksUri: `${env.AUTH_ISSUER}/.well-known/jwks.json`,
```
Comment (tpluscode) [inline apis/core/lib/auth.ts:23]: Reading this from `.well-known/openid-configuration` seeing how the JWKS endpoint has varying URL in oidc providers. So much or `.well-known` 🙄

**Comment by martinmaillard** [comment]
I get a 502 on `/env-config.js` when testing this PR

File: .local.env (hunk 1 above)
Comment (sandhose) [reply .local.env:16]: Yup; I think the instructions are up to date here: https://hackmd.io/rW_4v3BsRQS04DKDLKYpDQ#Full-stack-app-check-list

**Comment by tpluscode** [comment]
Can you try again @martinmaillard? 502 could mean that the API was not up yet (that is mostly coming from Lando's proxy)

**Comment by martinmaillard** [comment]
It may have been caused by a failure during build. I re-installed and rebuilt everything and it seems to work fine.

**Pull request merged**